pkg/storage/plugins/storage_protocol.go

```diff
@@ -121,11 +123,11 @@ type PatchOptions struct {
 
 	// Query is a query filter document
 	// See https://docs.mongodb.com/manual/core/document/#std-label-document-query-filter
-	QueryDocument interface{}
+	QueryDocument bson.M
 
 	// Transformation is set of instructions to modify matching
 	// documents.
-	Transformation interface{}
+	Transformation bson.D
```

Comment:
I switched to concrete types so that it was easier to write converters for the grpc client/server.